Ash.FrontEnd/src/js/global/components/Fido.jsx

```diff
@@ -0,0 +1,293 @@
+import React from 'react';
+import memoizeOne from 'memoize-one';
+
+/* EXAMPLE
+fetchItems = {
+  name: {
+    // ...configObject
+  },
+  nextName,
+  ...
+}
+*/
+
+/**
+ * Fetch configuration object default object shape
+ * @prop {string} path - fetch url
+ * @prop {object} query - object containing query string key-value pairs
+ * @prop {object} payload - json payload for request
+ * @prop {object} options - other options [optional]
+ * @prop {string} options.method - HTTP method type (case-insensitive) [optional]
+ * @prop {boolean} options.callOnMount - flag for whether the fetch should be
+ *    performed on mount [optional]
+ * @prop {boolean} options.preservePreviousData - flag for whether the fetch should
+ *    preserve the previously returned data set when a new call is initiated [optional]
+ */
+const defaultFetchConfig = {
+  path: '',
+  query: null,
+  payload: null,
+  options: {
+    method: 'get',
```

Comment:
Might think of exporting enums at some point instead of having strings for methods and other params like that that are easy to mistype. Fine for now though

Reply:
Good idea.  That's not hard to add.  Though, I do use a `.toUpperCase()`, so mistyping case won't matter.  Still think it's a good idea?

Reply:
It just makes it more clear what the options are (autocomplete is so nice), not really necessary (ideally enum translates to string so you support both)

Reply:
Alright, added!